yt/data_objects/construction_data_containers.py

```diff
@@ -2481,13 +2488,35 @@ def export_ply(
             elif sample_type == "vertex" and color_field not in self.vertex_samples:
                 self.get_data(color_field, sample_type, no_ghost=no_ghost)
         self._export_ply(
-            filename, bounds, color_field, color_map, color_log, sample_type
+            filename,
+            bounds,
+            color_field,
+            color_map,
+            color_log,
+            sample_type,
+            color_field_max=color_field_max,
+            color_field_min=color_field_min
         )
 
-    def _color_samples(self, cs, color_log, color_map, arr):
+    def _color_samples(self, cs, color_log, color_map, color_field_max, color_field_min, arr):
         if color_log:
             cs = np.log10(cs)
-        mi, ma = cs.min(), cs.max()
+        if color_field_min is None:
+            cs = [float(field) for field in cs]
+            cs = np.array(cs)
+            mi = cs.min()
```

Comment:
I'm not sure I understand this change. What's the gain here, compared to `mi = cs.min()` ?

Reply:
I'm not 100% (this is just a copy of the max/min treatment for the `export_obj` subroutine.  I'm guessing this gets rid of the units though?

Reply:
ah ! if dropping units is the intention here it should probably just be `mi = cs.ndview.min()`, which is not only more compact but also avoid copying data and conversions.

Reply:
I'm fairly baffled by this (and `export_obj`, so this comment isn't directed at you, @evanoconnor :) ). From what i can tell the input `cs` will always be an array? so `cs = [float(field) for field in cs]` a few lines above this will iterate over every element and individually cast to float just to convert back to an array afterwards? that seems not great.... am I missing something?

Reply:
I believe this is indeed to drop units, and *possibly* it's to do some conversion away from numpy arrays, because of some business with the exporting, but I cannot recall.

Reply:
`cs` is used a bit further down to construct the color scale from 0 to 1.  So I guess this then:

```
            cs = np.asarray(cs)
            mi = cs.min()
```

before deciding on the best way here, what is the general practice for units?  should the `color_field_max` and `color_field_max` have units at the onset, that would fail here because `cs` drops the units and `mi` and `ma` would retain them.

Reply:
yt's interface regarding scalar values with units isn't exactly consistent at the moment. In many places, we expect floats where we could in principle support passing quantities.
I think it's fine that `color_field_min` and `color_field_max` only support float values for now. We can always add support for quantities in the future.

Agreed that `cs = np.asarray(cs)` is the most robust and efficient way to drop units.